server/pipeline/create.go

```diff
@@ -77,15 +84,6 @@ func Create(ctx context.Context, _store store.Store, repo *model.Repo, pipeline
 		return nil, ErrFiltered
 	}
 
-	setGatedState(repo, pipeline)
-
-	err = _store.CreatePipeline(pipeline)
```

Comment:
did you look into if pipeline did got altered after creation - if so we have to update the db record here